src/main/scala/ai/privado/languageEngine/csharp/tagger/collection/CollectionTagger.scala

```diff
@@ -0,0 +1,210 @@
+    val controllerName = if annotation.code.contains("[controller]") then {
+      nodeType match
+        case NodeType.Class =>
+          annotation.start.typeDecl
+            .where(_.name(".*Controller"))
+            .name
+            .headOption
+            .getOrElse("")
+            .stripSuffix("Controller")
+            .toLowerCase
+
+        case NodeType.Method =>
+          annotation.start.method.typeDecl
+            .where(_.name(".*Controller"))
+            .name
+            .headOption
+            .getOrElse("")
+            .stripSuffix("Controller")
+            .toLowerCase
+    } else {
+      ""
+    }
+
+    // we get the action name. This can only be the method
+    val actionName = annotation.start.method.name.headOption.getOrElse("").toLowerCase
+
+    // Sometimes, we will get something like the following:
+    //
+    //    [Route("api/[controller]/[action]")]
+    //    public class SomeController : Controller {
+    //      [HttpGet]                                        <-- /api/some/copy           [CASE Ca]
+    //      public IActionResult Copy(){...}
+    //
+    //      [HttpGet("{id}")]                                <-- /api/some/paste/{id}     [CASE D]
+    //      public IActionResult Paste(int id){...}
+    //    }
+    //
+    //    [Route("api/[controller]")]
+    //    public class OtherController : Controller {
+    //      [HttpPost]                                        <-- /api/other/cut           [CASE Cb]
+    //      public IActionResult Cut(){...}
+    //    }
+
+    // We now try to build a URL under various combinations of annotations for Methods and Classes by doing token replacement. We try to cover most cases in the following guideline:
+    // https://learn.microsoft.com/en-us/aspnet/core/mvc/controllers/routing?view=aspnetcore-8.0#token-replacement-in-route-templates-controller-action-area
+    val url = nodeType match
+
+      case NodeType.Class =>
+        annotation.code
+          .replaceAll(".*\"(.*?)\".*", "/$1")
+          .replace("[controller]", controllerName)
+          .replace("[action]", actionName)
+
+      case NodeType.Method =>
+        // we are interested in cases where the method doesn't have action defined, but its containing type has
+        if annotation.start.method.typeDecl.annotation
+            .name(combinedRulePatterns)
+            .code
+            .headOption
+            .getOrElse("")
+            .matches(".*\\[action\\].*") && !annotation.code.matches(".*\\[action\\].*")
+        then {
+          if annotation.code.matches(".*\"(.*?)\".*") then {
+            s"$actionName" + annotation.code.replaceAll(
+              ".*\"(.*?)\".*",
+              "/$1"
+            ) // CASE D: we append action name for case like [HttpGet] where there is no argument
+          } else if !annotation.code.matches(".*\"(.*?)\".*") then {
+            s"$actionName" // CASE Ca: No brackets in annotation, means we don't need to append
+          } else {
```

Comment:
This looks unreachable, above we can have `if` and `else`, no need for `else if`

Reply:
Added `else if` to explicitly check for the case where we have and don't have `".*\"(.*?)\".*"` We might have other cases we will eventually have to support - in such cases, it would be nice just just send empty string for now instead of wrong `actionName` which might confuse us later.